cpp/src/fil/fil.cu

```diff
@@ -586,20 +586,35 @@ void tl2fil_common(forest_params_t* params, const tl::Model& model,
   // fill in forest-dependent params
   params->depth = max_depth(model);  // also checks for cycles
 
+  const tl::ModelParam& param = model.param;
+
   // assuming either all leaves use the .leaf_vector() or all leaves use .leaf_value()
   size_t leaf_vec_size = tl_leaf_vector_size(model);
   if (leaf_vec_size > 0) {
     ASSERT(leaf_vec_size == model.num_output_group,
            "treelite model inconsistent");
     params->num_classes = leaf_vec_size;
     params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
+    ASSERT(
+      param.pred_transform != "identity_multiclass",
+      "multiclass predict() only supported by choosing most likely label,\n"
+      "please use pred_transform == 'max_index' ('sigmoid' is still supported,"
+      " with 'max_index' implied)");
+    if ((std::string(param.pred_transform) == "max_index") &&
+        !(tl_params->output_class && tl_params->threshold == 0.5)) {
```

Comment:
For multi-class, `tl_params->output_class` should always be `true`, and the threshold is ignored, independent of the `pred_transform`.

So it's best to have a check like `ASSERT(tl_params->output_class, "output_class should be true for multi-class models")`.

Reply:
updated the assert statement to check if we are using a multi-class model and if so then `tl_params->output_class` should be true